open-metadata-implementation/view-services/glossary-author-view/glossary-author-view-server/src/main/java/org/odpi/openmetadata/viewservices/glossaryauthor/services/GlossaryAuthorViewGlossaryRESTServices.java

```diff
@@ -361,8 +364,57 @@ public GlossaryAuthorViewGlossaryRESTServices() {
             SubjectAreaNodeClients clients = instanceHandler.getSubjectAreaNodeClients(serverName, userId, methodName);
             Glossary glossary = clients.glossaries().restore(userId, guid);
             response.addResult(glossary);
-        }  catch (Throwable error) {
-            response =  getResponseForError(error, auditLog, className, methodName);
+        } catch (Throwable error) {
+            response = getResponseForError(error, auditLog, className, methodName);
+        }
+        restCallLogger.logRESTCallReturn(token, response.toString());
+        return response;
+    }
+    /**
+     * Create the supplied list of Terms in the glossary, identified by the supplied guid. Each term does not need to specify a glossary.
+     *
+     * @param serverName       local UI server name
+     * @param userId           user identifier
+     * @param guid             guid of the glossary under which the Terms will be created
+     * @param terms            terms to create
+     * @return a response which when successful contains a list of the responses from the Term creates (successful or otherwise). The order of the responses is the same as the supplied terms order.
+     *
+     * when not successful the following Exception responses can occur
+     * <ul>
+     * <li> UnrecognizedGUIDException            the supplied guid was not recognised</li>
+     * <li> UserNotAuthorizedException           the requesting user is not authorized to issue this request.</li>
+     * <li> FunctionNotSupportedException        Function not supported</li>
+     * <li> InvalidParameterException            one of the parameters is null or invalid.</li>
+     * <li> MetadataServerUncontactableException not able to communicate with a Metadata respository service.</li>
+     * </ul>
+     */
+    public SubjectAreaOMASAPIResponse<SubjectAreaOMASAPIResponse<Term>> createMultipleTermsInAGlossary(String serverName, String userId, String guid, Term[] terms) {
+        final String methodName = "createMultipleTermsInAGlossary";
+        RESTCallToken token = restCallLogger.logRESTCall(serverName, userId, methodName);
+
+        List<SubjectAreaOMASAPIResponse<Term>> termList = new ArrayList<>();
+        SubjectAreaOMASAPIResponse<SubjectAreaOMASAPIResponse<Term>> response = new SubjectAreaOMASAPIResponse<>();
+        AuditLog auditLog = null;
+
+        try {
+
+            SubjectAreaNodeClients clients = instanceHandler.getSubjectAreaNodeClients(serverName, userId, methodName);
+            for (Term term : terms) {
+                GlossarySummary glossarySummary = new GlossarySummary();
```

Comment:
I think not need this here because we already doing this action in `SubjectAreaTermRESTResource` on the server.

Reply:
The idea was that the terms that are submitted need not have the glossary set in them, so I could set it here. This is an API under a particular glossary so all the terms should be associated with that glossary, which is why I have done it this way.

Reply:
I think it is not good idea because bring additional responsibility to this method. He already implies that all terms will be created in this glossary...